### DISCUSSION TITLE
Implementing useSimulateContract without User Connection

### DISCUSSION THREAD

**Discussion by 0xSather** [wevm/wagmi#3892] [Q&A]
Hi everyone,

I am using the useSimulateContract hook to fetch swap quotes in Algebra using the quoteExactInputSingle function. I am looking to make it work even when the user is not connected.

This is my current code:

```
const quoteExactInputSingleCall = useSimulateContract({
  address: '0x94Ca5B835186A37A99776780BF976fAB81D84ED8', // algebra quoter v2
  abi: algebraQuoterV2ABI,
  functionName: 'quoteExactInputSingle',
  args: [
    {
      tokenIn: '0x4300000000000000000000000000000000000004', // WETH in Blast
      tokenOut: '0x4300000000000000000000000000000000000004', // USDB in Blast
      amountIn: 1000n,
      limitSqrtPrice: 0n,
    },
  ],
});
```

Currently, when a user is not connected, I face a ConnectorNotConnectedError.

How can I modify the implementation to use the Blast network as a fallback when the user is not connected? Any advice or example code on handling this situation would be greatly appreciated.

**Comment by 0xSather** [comment]
I'm using raimbowkit, wagmi and viem
```
"viem": "^2.9.15",
"wagmi": "^2.5.19",
"@rainbow-me/rainbowkit": "^2.0.1",
```

**Comment by tmm** [comment]
You need to explicitly pass an [`account`](https://wagmi.sh/react/api/hooks/useSimulateContract#account).

**Reply by 0xSather** [reply]
Thank you for the quick response and the helpful answer!
